File: model/barbearia.py
class Barbearia:
    def __init__(self, id, cnpj, nome, endereco, email, telefone, senha):
        self.id = id
        self.cnpj = cnpj
        self.nome = nome
        self.endereco = endereco
        self.email = email
        self.telefone = telefone
        self.senha = senha

    def __str__(self):
        return f"Barbearia(id = {self.id}, cnpj={self.cnpj}, nome={self.nome}, endereco={self.endereco}, email=${self.email}, telefone={self.telefone}, senha=${self.senha})"

File: backend/src/main/java/com/optimiza/clickbarber/model/Barbearia.java
package com.optimiza.clickbarber.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Table(name = "barbearias")
@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Barbearia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String cnpj;
    private String nome;
    private String endereco;
    private String telefone;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @OneToMany(mappedBy = "barbearia", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Servico> servicos;

    @OneToMany(mappedBy = "barbearia", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties("barbearia")
    private List<Barbeiro> barbeiros;

}
